File: src/com/androhi/androiddrawableviewer/form/ImageListCellRenderer.java
package com.androhi.androiddrawableviewer.form;

import com.intellij.ui.JBColor;

import javax.swing.*;
import java.awt.*;

public class ImageListCellRenderer implements ListCellRenderer {

    @Override
    public Component getListCellRendererComponent(JList list, Object value, int index, boolean isSelected, boolean cellHasFocus) {
        if (value instanceof JPanel) {
            setLabelColor(((JPanel) value), isSelected);
            Component component = (Component) value;
            Color normalBackground = index % 2 == 0 ? JBColor.border() : JBColor.background();
            component.setBackground(isSelected ? UIManager.getColor("List.selectionBackground") : normalBackground);
            return component;
        } else {
            return new JLabel("");
        }
    }

    private void setLabelColor(JPanel panel, boolean isSelected) {
        Component[] components = panel.getComponents();
        for (Component cmp : components) {
            cmp.setForeground(isSelected ? UIManager.getColor("List.selectionForeground") : JBColor.foreground());
        }
    }
}

File: src/main/kotlin/com/androhi/androiddrawableviewer/form/ImageListCellRenderer.kt
package com.androhi.androiddrawableviewer.form

import com.intellij.ui.JBColor
import java.awt.Component
import javax.swing.*

class ImageListCellRenderer : ListCellRenderer<Any> {

    override fun getListCellRendererComponent(list: JList<out Any>?, value: Any?, index: Int, isSelected: Boolean, cellHasFocus: Boolean): Component =
        when (value) {
            is JPanel -> {
                setLabelColor(value, isSelected)
                val normalColor = if (index % 2 == 0) JBColor.border() else JBColor.background()
                val selectedColor = UIManager.getColor("List.selectionBackground")
                value.apply {
                    background = if (isSelected) selectedColor else normalColor
                }
            }
            else -> {
                JLabel("")
            }
        }

    private fun setLabelColor(panel: JPanel, isSelected: Boolean) =
            panel.components.forEach {
                it.foreground = if (isSelected) {
                    UIManager.getColor("List.selectionForeground")
                } else {
                    JBColor.foreground()
                }
            }
}